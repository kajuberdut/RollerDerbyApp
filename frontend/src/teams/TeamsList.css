.Teams {
    margin-top: 150px;
}

.TeamsList {
    background-color: #d1d2d4;
    margin-top: 20px;
    width: 43%;
    margin-left: 12%;
}