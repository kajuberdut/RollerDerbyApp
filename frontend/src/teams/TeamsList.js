import React, { useState, useEffect } from "react";
import FastApi from "../Api";
import Loading from "../multiUse/loading/Loading"
import './TeamsList.css'
import { Card, CardBody, CardTitle } from "reactstrap";
import TeamComponent from "./TeamComponent";
import InvitesList from "../invites/InvitesList";

/**
 * Display teams page
 */

function TeamsList({ getPendingInvites }) {

  /** Set Teams and is loading in state*/

  const [isLoading, setIsLoading] = useState(true);
  const [teams, setTeams] = useState([]);

   /** Retrieve user from local storage */

  const user = JSON.parse(localStorage.getItem('user'));

  /** API get request for teams */

  async function getTeams() {

    try {

      let teams = await FastApi.getTeams(user.userId);
      console.log("teams:", teams)
      setTeams(teams)
      setIsLoading(false)
    } catch (errors) {

      return { success: false, errors };
    }
  }

  /** Reloads teams when changes request for teams */

  useEffect(() => {
      getTeams();
  }, []);

  /** Display loading if API call is has not returned */

    if (isLoading) {
      return (
          <Loading />
      )
    }

  /** Render the cards for users*/

    const renderCards = () => {
      return (
        <div className="TeamList-RenderCards">
                {teams.map(team => (
                  <TeamComponent team={team} key={"Team-" + team.groupId} />           
                ))}
          </div>
        );
    }
  
  /** Render search bar and cards */

  return (
      <div className="Teams">
        <div>
          <a href='/teams/add'>
            <button style={{borderRadius: '4px'}}>Create Team</button>
          </a>
        </div>

        <div style={{display: 'flex'}}>
        <Card className="TeamsList" style={{minWidth: '400px'}}>
            <CardBody>
              <CardTitle><h1 style={{paddingBottom: '30px'}}>Teams</h1></CardTitle>
              {teams && <div> {renderCards()} </div> }
              {teams.length === 0 && <div>Sorry, you are not on any teams.</div>}
            </CardBody>
        </Card>

        <InvitesList getTeams={getTeams} getPendingInvites={getPendingInvites}/>

        </div>
      </div>
  );
}

export default TeamsList;