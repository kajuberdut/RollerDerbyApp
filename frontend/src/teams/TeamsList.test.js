import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import TeamsList from './TeamsList'; 
import FastApi from '../Api'; 
import { BrowserRouter } from "react-router-dom";

// * Passing as of 2/15/24
// npm test TeamsList.test.js
// must be in frontend directory

// Mock FastApi methods
jest.mock('../Api', () => ({
  getTeams: jest.fn(),
}));

// Mock useParams
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
}));

describe('TeamList Component', () => {
    const mockTeams = [
        { 
            groupId: 1,
            name: "Black and Blue",
        },
        { 
            groupId: 2,
            name: "Bold Bashers",
        }
    ]

  beforeEach(() => {
    FastApi.getTeams.mockResolvedValue(mockTeams);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('fetches teams and displays page details', async () => {
    render(
    <BrowserRouter>
        <TeamsList />
    </BrowserRouter>
    );

    expect(screen.getByTestId('loading-indicator')).toBeInTheDocument();

    // Wait for the API call to resolve and check if the details are displayed
    await waitFor(() => {
        expect(screen.getByText('Create Team')).toBeInTheDocument();
        expect(screen.getByText('Teams')).toBeInTheDocument();
    });
  });

  it('fetches teams and displays team names', async () => {
    render(
        <BrowserRouter>
            <TeamsList />
        </BrowserRouter>
        );

    expect(screen.getByTestId('loading-indicator')).toBeInTheDocument();

    // Wait for the API call to resolve and check if the details are displayed
    await waitFor(() => {
      expect(screen.getByText(mockTeams[0].name)).toBeInTheDocument();
      expect(screen.getByText(mockTeams[1].name)).toBeInTheDocument();
    });

  });

  test('matches snapshot', function() {
    const { asFragment } = render(<TeamsList />);
    expect(asFragment()).toMatchSnapshot(); 
  })

});

