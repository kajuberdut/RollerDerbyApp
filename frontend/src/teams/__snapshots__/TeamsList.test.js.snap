// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamDetails Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Loading"
    data-testid="loading-indicator"
  />
</DocumentFragment>
`;

exports[`TeamList Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Loading"
    data-testid="loading-indicator"
  />
</DocumentFragment>
`;
